packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -522,15 +520,45 @@ export class StripeHelper {
       },
       { idempotencyKey: `ssc-${subIdempotencyKey}` }
     );
-    if (promotionCode && subscription.discount?.promotion_code) {
-      subscription.discount.promotion_code = promotionCode;
+
+    const updatedSubscription = await this.postSubscriptionCreationUpdates({
+      subscription,
+      promotionCode,
+    });
+
+    return updatedSubscription;
+  }
+
+  private async postSubscriptionCreationUpdates({
+    subscription,
+    promotionCode,
+  }: {
+    subscription: Stripe.Response<Stripe.Subscription>;
+    promotionCode?: Stripe.PromotionCode;
+  }) {
+    // Save the promotion code into the subscription's metadata now that the
+    // subscription has been successfully created.
+    if (
+      promotionCode &&
+      (subscription.latest_invoice as Stripe.Invoice).discount
+    ) {
+      const subscriptionMetadata = {
+        ...subscription.metadata,
+        [SUBSCRIPTION_PROMOTION_CODE_METADATA_KEY]: promotionCode.code,
+      };
+      subscription.metadata = subscriptionMetadata;
```

Comment:
Nit: Can we do the assignment at the same time?

Reply:
I'm certainly not against it.  But now that CI is passing I don't want to run it again for just this change.  The assigned var is used below in the update call as well fwiw.